crates/nu-command/src/platform/is_terminal.rs

```diff
@@ -0,0 +1,90 @@
+use nu_protocol::{
+    ast::{Argument, Call},
+    engine::{Command, EngineState, Stack},
+    Category, PipelineData, ShellError, Signature, Span, Type, Value,
+};
+use std::io::IsTerminal as _;
+
+#[derive(Clone)]
+pub struct IsTerminal;
+
+impl Command for IsTerminal {
+    fn name(&self) -> &str {
+        "is-terminal"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("is-terminal")
+            .input_output_type(Type::Nothing, Type::Bool)
+            .switch("stdin", "Check if stdin is a terminal", Some('i'))
+            .switch("stdout", "Check if stdout is a terminal", Some('o'))
+            .switch("stderr", "Check if stderr is a terminal", Some('e'))
+            .category(Category::Platform)
+    }
+
+    fn usage(&self) -> &str {
+        "Check if stdin, stdout, or stderr is a terminal"
+    }
+
+    fn run(
+        &self,
+        _engine_state: &EngineState,
+        _stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        let PipelineData::Empty = input else {
+            let src_span = input.span().unwrap_or(Span::unknown());
+            return Err(ShellError::PipelineMismatch { exp_input_type: "nothing".into(), dst_span: call.head, src_span  });
+        };
+
+        let stdin = call.has_flag("stdin");
+        let stdout = call.has_flag("stdout");
+        let stderr = call.has_flag("stderr");
+
+        let is_terminal = match (stdin, stdout, stderr) {
+            (true, false, false) => std::io::stdin().is_terminal(),
+            (false, true, false) => std::io::stdout().is_terminal(),
+            (false, false, true) => std::io::stderr().is_terminal(),
+            (false, false, false) => {
+                return Err(ShellError::MissingParameter {
+                    param_name: "one of --stdin, --stdout, --stderr".into(),
+                    span: call.head,
+                });
+            }
+            _ => {
+                let span = match call.arguments.len() {
+                    0 => call.span(),
+                    1 => *argument_span(call.arguments.first().expect("at least one argument")),
+                    _ => {
+                        // Build a span covering all arguments.  At least one of them must be removed
+                        let first =
+                            argument_span(call.arguments.first().expect("at least one argument"));
+                        let last =
+                            argument_span(call.arguments.last().expect("at least one argument"));
+
+                        Span::new(first.start, last.end)
+                    }
+                };
```

Comment:
My spidey-sense it tingling that this part is just a little weird. I've looked at a lot of nushell code and haven't seen people do this before. I'm not sure what to make of it.

Reply:
See #10983 which elevates this to a pattern with a better implementation. I want to have nice error messages like this:

```
❯ is-terminal --stdin --stdout --stderr
Error: nu::shell::incompatible_parameters

  × Incompatible parameters.
   ╭─[source:1:1]
 1 │ is-terminal --stdin --stdout --stderr
   ·             ────────────┬────────────
   ·                         ╰── Only one stream may be checked
   ╰────
```

One of them was wrong, but which one?

Building in a behavior expectation around multiple arguments and documenting it seems worse than saying "just one of these, please" with proper underlining.

Reply:
There's `ShellError::IncompatibleParameters` for two; the "proper" solution is probably to have an N version. Handing off individual spans instead of one merged span then also supports the case where there are other params between the incompatible ones.

Reply:
I could use `IncompatibleParameters`, but would like to have at least the `Argument::span()` from #10983 to make it look nicer in here in #10970, along with a way to combine spans. The functions currently available make constructing this kind of error clunky.

Reply:
I've updated this to remove the weirdness by taking advantage of `Argument::span()`

I'm still using `IncompatibleParametersSingle` because making a reasonable message for `IncompatibleParameters` seems like a bunch of work when we don't know which argument is the one the user most-wants to check for.

Reply:
Thanks